### DISCUSSION TITLE
../ucrt/time.h No such file or directory

### DISCUSSION THREAD

**Discussion by oatless** [musescore/MuseScore#17175] [Q&A] [answered]
Trying to compile MuseScore on Windows using Qt 5.15.2 and MinGW 64-bit.

For some reason, I get errors similar to the title of the question. 

Tried installing the newest Windows SDK and checking that ucrt/time.h actually exists and specified all the environment variables as detailed by https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-windows-visual-studio/compile-0

Still, I am not able to compile with Qt 5.15.2.

Does anyone have any ideas on what the problem might be?

Thanks in advance.

**Comment by cbjeukendrup** [comment, marked as answer]
Have you tried using MSVC instead of MinGW? More people say that it doesn't compile with MinGW anymore.

**Reply by oatless** [reply]
Gonna try this right now.

**Comment by Jojo-Schmitz** [comment]
At least the new sound system seems incompatible with MinGW anymore, but that particular error message is new to me

**Comment by oatless** [comment]
Following https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-windows-visual-studio/compile

Compiling with Qt Creator with Qt 5,12.2 MSVC2019 64-bit kit was successful.

**Comment by oatless** [comment]
For further reference here is the full list of errors I am getting when compiling with MinGW 64-bit:

../ucrt/sys/types.h: No such file or directory
../ucrt/time.h No such file or directory
winrt/Windows.Devices.Enumeration.h: No such file or directory

**Comment by Jojo-Schmitz** [comment]
The winrt/Windows.Devices.Enumeration.h stuff is what I get too, not the others though

**Reply by oatless** [reply]
Both "../ucrt/sys/types.h" and "../ucrt/time.h" errors seem to be coming from google_crash_client\compat\win\sys and google_crash_client\compat\win respectively

In both header files there is an include referencing <../ucrt/sys/types.h> or <../ucrt/time.h> which is causing the error.

**Reply by Jojo-Schmitz** [reply]
Ah, yes, I believe I configured that to not build when using MinGW
